exasol_transformers_extension/udfs/question_answering_udf.py

```diff
@@ -16,12 +17,17 @@ def __init__(self,
         self.pipeline = pipeline
         self.base_model = base_model
         self.tokenizer = tokenizer
+        self.device = None
         self.cache_dir = None
         self.last_loaded_model_name = None
         self.last_loaded_model = None
         self.last_loaded_tokenizer = None
 
     def run(self, ctx):
+        device_name = ctx.get_dataframe(1).iloc[0]['device_name']
+        self.device = torch.device(device_name.lower())
```

Comment:
use `device = torch.device(f"cuda:{device_name}" if torch.cuda.is_available() and device_name is not None  else "cpu")`

For the moment, we only support cuda. Should this change, we probably should use detection. I would like to hide the details from the user as much as possible, Maybe at some point, the database might inject the device to use, instead the user need to specify it. 

I also suggest that we implement a function, which returns the number of GPU devices, because we probably need it in the queries to generate device ids we call the PredictionUDFs with.